client/src/app/core/routing/menu-guard.service.ts

```diff
@@ -0,0 +1,45 @@
+import { Injectable } from '@angular/core'
+import { CanActivate, CanDeactivate } from '@angular/router'
+import { MenuService } from '@app/core/menu'
+import { ScreenService } from '@app/shared/misc/screen.service'
+
+@Injectable()
+export class MenuGuards {
+  public static guards: any = []
+
+  static open () {
+    MenuGuards.guards.push(OpenMenuGuard)
```

Comment:
Why do you need to push `OpenMenuGuard`?

Reply:
To make the providers in the module definition properly consider them.

Reply:
Okay then please use a `Set` or check duplicates so you don't add multiple times the same guard